Date: Sat, 4 Mar 2006 00:40:28 -0500
From: Aron Griffis <agriffis gentoo org>
To: svk-dev lists openfoundry org
Subject: patch for SVK/Resolve/XXDiff.pm
Message-ID: <20060304054028.GA3958@olive.flatmonk>

The following patch fixes svk smerge with xxdiff.  Here is the
problem, from xxdiff --help:

  --exit-with-merge-status, -X  If all diff hunks are selected and no
                                unsaved selections exist, then exit
                                with code of 0.  Normally, xxdiff will
                                pass back the diff return code. 

Since diff returns an exit code of 1 when there are differences, svk
believes that the merge always fails.

--- SVK-1.07/lib/SVK/Resolve/XXDiff.pm.old	2005-03-28 20:46:18.000000000 -0500
+++ SVK-1.07/lib/SVK/Resolve/XXDiff.pm	2006-03-04 00:39:53.000000000 -0500
@@ -5,7 +5,7 @@
 sub arguments {
     my $self = shift;
     return (
-        qw( -m -O -M ),
+        qw( -X -m -O -M ),
         @{$self}{qw( merged yours base theirs )}
     );
 }

